crates/plex-api/src/media_container/server/library.rs

```diff
@@ -0,0 +1,496 @@
+    #[serde(flatten)]
+    pub grand_parent: GrandParentMetadata,
+    #[serde(rename = "librarySectionID")]
+    pub library_section_id: Option<u32>,
+    pub library_section_title: Option<String>,
+    #[serde(rename = "librarySectionUUID")]
+    pub library_section_uuid: Option<String>,
+    pub media_tag_prefix: Option<String>,
+    #[serde(default, with = "time::serde::timestamp::option")]
+    pub media_tag_version: Option<OffsetDateTime>,
+    pub sort_asc: Option<bool>,
+    pub thumb: Option<String>,
+    pub title1: Option<String>,
+    pub title2: Option<String>,
+    pub summary: Option<String>,
+    pub theme: Option<String>,
+    pub view_group: Option<String>,
+    pub view_mode: Option<u32>,
+    #[serde(flatten)]
+    pub media_container: MediaContainer,
+    #[serde(default, rename = "Directory")]
+    pub directories: Vec<Value>,
+    #[serde(default, rename = "Metadata")]
+    pub metadata: Vec<Metadata>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub enum PivotType {
+    Hub,
+    List,
+    View,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct Pivot {
+    pub context: String,
+    pub id: String,
+    pub key: String,
+    pub symbol: String,
+    pub title: String,
+    #[serde(rename = "type")]
+    pub pivot_type: PivotType,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum LibraryType {
+    Movie,
+    Show,
+    Artist,
+    Photo,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct ServerLibrary {
+    #[serde(rename = "type")]
+    pub media_type: LibraryType,
+    #[serde(rename = "Pivot")]
+    pub pivots: Vec<Pivot>,
+    pub agent: String,
+    pub hub_key: String,
+    pub id: String,
+    pub key: String,
+    pub subtype: Option<String>,
+    pub language: String,
+    pub refreshing: bool,
+    #[serde(with = "time::serde::timestamp")]
+    pub scanned_at: OffsetDateTime,
+    pub scanner: String,
+    pub title: String,
+    #[serde(with = "time::serde::timestamp")]
+    pub updated_at: OffsetDateTime,
+    pub uuid: String,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct ServerPlaylists {
+    #[serde(rename = "type")]
+    _type: MustBe!("playlist"),
+    #[serde(rename = "Pivot")]
+    pub pivots: Vec<Pivot>,
+    pub id: String,
+    pub key: String,
+    pub title: String,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct ServerHome {
+    pub hub_key: String,
+    pub title: String,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(untagged)]
```

Comment:
If only ServerHome had some kind of a tag :( I don't trust the flexibility of the untagged enums, especially when working with Plex API :D